server/src/main/java/org/apache/cassandra/sidecar/handlers/OpenApiHandler.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.handlers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.cassandra.sidecar.config.OpenApiConfiguration;
+import org.apache.cassandra.sidecar.config.SidecarConfiguration;
+
+/**
+ * Handler that serves the OpenAPI specification
+ * 
+ * This handler serves the OpenAPI specification generated by the Gradle openApiGenerate task
+ * which processes JAX-RS/MicroProfile annotations to create comprehensive API documentation.
+ */
+@Singleton
+public class OpenApiHandler implements Handler<RoutingContext>
+{
+    private static final Logger logger = LoggerFactory.getLogger(OpenApiHandler.class);
+    private final OpenApiConfiguration openApiConfig;
```

Comment:
This field is no longer used. And the classes `OpenApiConfiguration` and `OpenApiConfigurationImpl` can both be removed.